app/code/Magento/Sales/Ui/Component/Listing/Column/PurchasedPrice.php

```diff
@@ -62,7 +72,9 @@ public function prepareDataSource(array $dataSource)
     {
         if (isset($dataSource['data']['items'])) {
             foreach ($dataSource['data']['items'] as & $item) {
-                $currencyCode = isset($item['order_currency_code']) ? $item['order_currency_code'] : null;
+                $currencyCode = isset($item['order_currency_code'])
+                    ? $item['order_currency_code']
+                    : $this->order->get($item['order_id'])->getOrderCurrencyCode();
```

Comment:
1. Looks like here we can use `??` here

```suggestion
                $currencyCode = $item['order_currency_code'] ?? $this->order->get($item['order_id'])->getOrderCurrencyCode();
```

2. Loading objects in the loop is a really heavy operation and not recommended. Can we load all needed orders at once? Or maybe we can do that when preparing dataSource?

Reply:
1. Thanks.
2. I think no. Different orders may have different currencies